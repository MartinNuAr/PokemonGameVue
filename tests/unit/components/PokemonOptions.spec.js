import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions"
import { pokemon } from "../mocks/pokemon.mock";

describe('PokemonOptions component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemonGroup: pokemon
            }
        })
    })

    test('Debe hacer match con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('Debe mostrar las 4 opciones correctamente', () => {

        const liElements = wrapper.findAll('li')

        expect(liElements.length).toBe(4)

        expect(liElements[0].text()).toBe('Pikachu')
        expect(liElements[1].text()).toBe('Charmander')
        expect(liElements[2].text()).toBe('Venusaur')
        expect(liElements[3].text()).toBe('Mew')

    })

    test('De emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => {

        const [li1, li2, li3, li4] = wrapper.findAll('li')

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect(wrapper.emitted('selection').length).toBe(4)

        expect(wrapper.emitted('selection')[0]).toEqual([5])
        expect(wrapper.emitted('selection')[1]).toEqual([10])
        expect(wrapper.emitted('selection')[2]).toEqual([15])
        expect(wrapper.emitted('selection')[3]).toEqual([20])

    })

})
