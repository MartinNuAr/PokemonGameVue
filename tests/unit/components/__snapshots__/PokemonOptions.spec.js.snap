// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>Pikachu</li>
    <li>Charmander</li>
    <li>Venusaur</li>
    <li>Mew</li>
  </ul>
</div>
`;
